src/com/facebook/buck/apple/toolchain/impl/AppleToolchainDiscovery.java

```diff
@@ -73,7 +76,12 @@ private AppleToolchainDiscovery() {}
 
       try (DirectoryStream<Path> toolchainStream =
           Files.newDirectoryStream(toolchains, "*.xctoolchain")) {
-        for (Path toolchainPath : toolchainStream) {
+
+        List<Path> sortedToolchainPaths = new ArrayList<>();
+        toolchainStream.forEach(sortedToolchainPaths::add);
+        sortedToolchainPaths.sort(Comparator.comparing(Path::toString));
```

Comment:
It's been a while since I wrote this patch but I do remember that iteration order would affect which toolchain buck would pick given that Files.newDirectoryStream() doesn't have a well defined iteration order. Having said that, I can't reproduce the problem with the current master.

I only kept this to minimize the number of changes we had to do in `AppleToolchainDiscoveryTest.java` due to the usage of `ImmutableMap`s.